### PR TITLE
pythonPackages.asf_search: init at 3.0.6

### PR DESCRIPTION
**Pull request by bzizou**
###### Motivation for this change
python module used by our HPC users

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -628,6 +628,8 @@ in {
 
   ase = callPackage ../development/python-modules/ase { };
 
+  asf_search = callPackage ../development/python-modules/asf_search { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:631]:
```suggestion
  asf-search = callPackage ../development/python-modules/asf-search { };
```
Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)

File: pkgs/development/python-modules/asf_search/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "3.0.6";
+
+  src = fetchFromGitHub {
+    owner = "asfadmin";
+    repo = "Discovery-${pname}";
+    rev = "v${version}";
+    sha256 = "1jzah2l1db1p2mv59w9qf0x3a9hk6s5rzy0jnp2smsddvyxfwcyn";
+  };
+
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/asf_search/default.nix:14]: Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

File: pkgs/development/python-modules/asf-search/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  version = "3.0.6";
+
+  src = fetchFromGitHub {
+    owner = "asfadmin";
+    repo = "Discovery-${pname}";
+    rev = "v${version}";
+    sha256 = "1jzah2l1db1p2mv59w9qf0x3a9hk6s5rzy0jnp2smsddvyxfwcyn";
+  };
+
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/asf-search/default.nix:14]: Can we execute the [tests](https://github.com/asfadmin/Discovery-asf_search/tree/master/tests)?

Comment (bzizou) [reply pkgs/development/python-modules/asf-search/default.nix:14]: Not for now. As far as I understand, we should wait for 3.1.0 for a working test workflow: https://github.com/asfadmin/Discovery-asf_search/blob/master/CHANGELOG.md

File: pkgs/development/python-modules/asf-search/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pytz, shapely, importlib-metadata, requests, python-dateutil }:
+
+buildPythonPackage rec {
+  pname = "asf_search";
+  version = "3.0.6";
+
+  src = fetchFromGitHub {
+    owner = "asfadmin";
+    repo = "Discovery-${pname}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/asf-search/default.nix:9]:
```suggestion
    repo = "Discovery-asf_search";
```

File: pkgs/development/python-modules/asf-search/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "v${version}";
+    sha256 = "1jzah2l1db1p2mv59w9qf0x3a9hk6s5rzy0jnp2smsddvyxfwcyn";
+  };
+
+  doCheck = false;
+
+  pythonImportsCheck = [ "asf_search" ];
+
+  propagatedBuildInputs = [ pytz shapely importlib-metadata requests python-dateutil ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/asf-search/default.nix:19]:
```suggestion
```

File: pkgs/development/python-modules/asf-search/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+buildPythonPackage rec {
+  pname = "asf_search";
+  version = "3.0.6";
+
+  src = fetchFromGitHub {
+    owner = "asfadmin";
+    repo = "Discovery-${pname}";
+    rev = "v${version}";
+    sha256 = "1jzah2l1db1p2mv59w9qf0x3a9hk6s5rzy0jnp2smsddvyxfwcyn";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/asf-search/default.nix:12]:
```suggestion
  };
  
    propagatedBuildInputs = [ pytz shapely importlib-metadata requests python-dateutil ];

```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.asf_search

**Pull request merged**